src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -481,27 +380,7 @@ private void VerifyCanWriteStartChangeset(bool synchronousCall)
         /// <summary>
         /// Starts a new changeset - implementation of the actual functionality.
         /// </summary>
-        private void WriteStartChangesetImplementation()
-        {
-            // write pending message data (headers, response line) for a previously unclosed message/request
-            this.WritePendingMessageData(true);
-
-            // important to do this first since it will set up the change set boundary.
-            this.SetState(BatchWriterState.ChangeSetStarted);
-            Debug.Assert(this.changeSetBoundary != null, "this.changeSetBoundary != null");
-
-            // reset the size of the current changeset and increase the size of the batch
-            this.ResetChangeSetSize();
-            this.InterceptException(this.IncreaseBatchSize);
-
-            // write the boundary string
-            ODataBatchWriterUtils.WriteStartBoundary(this.rawOutputContext.TextWriter, this.batchBoundary, !this.batchStartBoundaryWritten);
-            this.batchStartBoundaryWritten = true;
-
-            // write the change set headers
-            ODataBatchWriterUtils.WriteChangeSetPreamble(this.rawOutputContext.TextWriter, this.changeSetBoundary);
-            this.changesetStartBoundaryWritten = false;
-        }
+        internal abstract void WriteStartChangesetImplementation();

```

Comment:
Implementation methods are not generally in base classes, they are called from the derived classe's override mehtod. -- is there a reason to have this here?